benchopt/plotting/html/templates/result.mako.html

```diff
@@ -169,10 +170,37 @@ <h1>Result on ${benchmark} benchmark</h1>
 
   var obj = document.getElementById('dataset_selector');
   obj.onchange();
-  var obj = document.getElementById('objective_selector');
+  var prev_obj_selector = obj.value;
+  var obj = document.getElementById('objective_column');
   obj.onchange();
+  var prev_obj_col = obj.value;
   var obj = document.getElementById('plot_kind');
   obj.onchange();
+  var prev_kind = obj.value;
+
+  function set_visibility() {
+    var kind = document.getElementById('plot_kind').value;
+    var obj_select = document.getElementById("objective_selector").value;
+    var obj_col = document.getElementById("objective_column").value;
+    if(kind !== "histogram"){
+      if(prev_kind != null){
+        // old graph values
+        var graph_id = document.querySelector('[id$="' + prev_obj_selector + prev_obj_col + prev_kind + '"]').id;
+        var graph = document.getElementById(document.getElementById(graph_id).getElementsByTagName("div")[1].id);
+        var traces_visib = graph.data.map(trace => trace.visible);
+        // new graph values
+        var graph_id = document.querySelector('[id$="' + obj_select + obj_col + kind + '"]').id;
+        var graph = document.getElementById(document.getElementById(graph_id).getElementsByTagName("div")[1].id);
+        for(i=0; i < graph.data.length; i++){
+          graph.data[i].visible = traces_visib[i];
+        }
+        Plotly.redraw(graph);
+      }
+      prev_kind = kind;
```

Comment:
A suggestion: maybe store the graph_id so it is simpler code to read?
Not mandatory of course.

```suggestion
      // Store the current graph id so we can update the visible traces on the next change.
      prev_kind = kind;
```